test/ct_test.c

```diff
@@ -486,19 +491,34 @@ static int test_encode_tls_sct()
     SETUP_CT_TEST_FIXTURE();
 
     SCT *sct = SCT_new();
-    SCT_set_version(sct, 0);
-    SCT_set1_log_id(sct, (unsigned char *)
+    if (!SCT_set_version(sct, 0)) {
+        fprintf(stderr, "Failed to set SCT version\n");
+        return 1;
+    }
+    if (!SCT_set1_log_id(sct, (unsigned char *)
         "\xDF\x1C\x2E\xC1\x15\x00\x94\x52\x47\xA9\x61\x68\x32\x5D\xDC\x5C\x79"
-        "\x59\xE8\xF7\xC6\xD3\x88\xFC\x00\x2E\x0B\xBD\x3F\x74\xD7\x64", 32);
+        "\x59\xE8\xF7\xC6\xD3\x88\xFC\x00\x2E\x0B\xBD\x3F\x74\xD7\x64", 32)) {
+        fprintf(stderr, "Failed to set SCT log ID\n");
+        return 1;
+    }
     SCT_set_timestamp(sct, 1);
-    SCT_set1_extensions(sct, (unsigned char *)"", 0);
-    SCT_set_signature_nid(sct, NID_ecdsa_with_SHA256);
-    SCT_set1_signature(sct, (unsigned char *)
+    if (!SCT_set1_extensions(sct, (unsigned char *)"", 0)) {
```

Comment:
`SCT_set1_extensions` should, perhaps, be permissive of `NULL, 0`, if it isnt.

Is there a semantic difference between `NULL` extensions and empty extensions? Shouldn't be, right - extensions can't be missing, they can just be empty.